app/models/member.rb

```diff
@@ -86,4 +88,10 @@ def token_payload
       authentication_id: authentication.try(:id)
     }
   end
+
+  private
+
+  def existing_memberships_and_email_changed?
+    Member.where(email: email).where.not(id: id).any? && email_changed?
```

Comment:
I understand that removing duplicates might be a complex task, but in the mean we could achieve the exact same behaviour by validating uniqueness only `on: create`.